### PR TITLE
LabelWithOptions - Add checkbox

### PR DESCRIPTION
**Pull request by danielstro**
- Added the option to show checkboxes next to the options.
- Fixed a problem with checkbox that causes multiple events to be fired

### CODE REVIEW

File: packages/wix-ui-core/src/components/Checkbox/Checkbox.tsx [hunk 1]
```diff
@@ -47,6 +47,7 @@ export class Checkbox extends React.Component<CheckboxProps, CheckboxState> {
           <input
             type="checkbox"
             className={styles.nativeCheckbox}
+            onClick={e => e.stopPropagation()}
```
Comment (shlomitc) [inline packages/wix-ui-core/src/components/Checkbox/Checkbox.tsx]: why is this needed?
probably need a test for it

File: packages/wix-ui-core/src/components/LabelWithOptions/LabelWithOptions.driver.ts [hunk 2]
```diff
@@ -1,23 +1,28 @@
...
 
   return Object.assign({}, dropdownDriver, labelDriver, {
     getSuffix: () => suffixElement,
-    isRequired: () => domUtils.hasStyleState(args.element, 'required'),
-    isError: () => domUtils.hasStyleState(args.element, 'error'),
-    isDisabled: () => domUtils.hasStyleState(args.element, 'disabled')
+    isRequired: () => domUtils.hasStyleState(element, 'required'),
+    isError: () => domUtils.hasStyleState(element, 'error'),
+    isDisabled: () => domUtils.hasStyleState(element, 'disabled'),
+    checkboxAt: (index: number) => {
```
Comment (shlomitc) [inline packages/wix-ui-core/src/components/LabelWithOptions/LabelWithOptions.driver.ts]: driver function name should make it clear that it's a driver that returned

File: packages/wix-ui-core/src/components/LabelWithOptions/LabelWithOptions.st.css [hunk 3]
```diff
@@ -15,12 +25,18 @@
 
 .root {
   -st-extends: Dropdown;
-  -st-states: required, error, disabled;
+  -st-states: required, error, disabled, checkbox;
 }
 
+@custom-selector :--selection .root:checkbox DropdownOption:selected;
```
Comment (shlomitc) [inline packages/wix-ui-core/src/components/LabelWithOptions/LabelWithOptions.st.css]: the name `selection` might be too general

File: packages/wix-ui-core/src/components/LabelWithOptions/LabelWithOptions.st.css [hunk 4]
```diff
@@ -30,4 +46,25 @@
 }
 
 .placeholder {
+}
+
+.optionCotainer {
+  display: flex;
+}
+
+.checkbox {
```
Comment (shlomitc) [inline packages/wix-ui-core/src/components/LabelWithOptions/LabelWithOptions.st.css]: why do much styling is needed?

File: packages/wix-ui-core/src/components/LabelWithOptions/LabelWithOptions.tsx [hunk 5]
```diff
@@ -127,42 +97,84 @@ export class LabelWithOptions extends React.PureComponent<LabelWithOptionsProps,
...
+    if (!this.props.checkbox) {
+      return this.props.options;
+    }
+    return this.props.options.map(option => {
+      let newOption = {id: option.id, isDisabled: option.isDisabled, isSelectable: option.isSelectable, value: option.value, render: null};
+      const checked = this.state.selectedIds.includes(option.id);
+      newOption.render = option.isSelectable ?
+                         value => (
+                          <div className={style.optionCotainer}>
+                            <Checkbox disabled={option.isDisabled} checked={checked} className={style.checkbox}/>{option.render(value)}
```
Comment (shlomitc) [inline packages/wix-ui-core/src/components/LabelWithOptions/LabelWithOptions.tsx]: a disabled checkbox still shows a "hand" mouse, probably should be fixed in the `Checkbox` component

Comment (danielstro) [reply packages/wix-ui-core/src/components/LabelWithOptions/LabelWithOptions.tsx]: This is an issue with the checkbox

File: packages/wix-ui-core/src/components/Checkbox/Checkbox.tsx (hunk 1 above)
Comment (danielstro) [reply packages/wix-ui-core/src/components/Checkbox/Checkbox.tsx]: This is one of the things that we can't really test because of the onChange problem. Because we don't really use a browser, we can't recreate this.

File: packages/wix-ui-core/src/components/LabelWithOptions/LabelWithOptions.driver.ts (hunk 2 above)
Comment (danielstro) [reply packages/wix-ui-core/src/components/LabelWithOptions/LabelWithOptions.driver.ts]: I'm not sure what name to give it. checkboxDriverAt?

File: packages/wix-ui-core/src/components/LabelWithOptions/LabelWithOptions.st.css (hunk 4 above)
Comment (danielstro) [reply packages/wix-ui-core/src/components/LabelWithOptions/LabelWithOptions.st.css]: I removed it to the bare minimum needed to actually see something in the storybook

File: packages/wix-ui-core/src/components/Checkbox/Checkbox.tsx (hunk 1 above)
Comment (shlomitc) [reply packages/wix-ui-core/src/components/Checkbox/Checkbox.tsx]: but shouldn't this be the responsibility of the container that holds the checkbox?

Comment (danielstro) [reply packages/wix-ui-core/src/components/Checkbox/Checkbox.tsx]: I'm not sure, in this case when someone is using a checkbox they don't expect to receive two calls for their onChange handler, so they will not know to disable this.

Comment (shlomitc) [reply packages/wix-ui-core/src/components/Checkbox/Checkbox.tsx]: ok

File: packages/wix-ui-core/src/components/LabelWithOptions/LabelWithOptions.st.css (hunk 4 above)
Comment (shlomitc) [reply packages/wix-ui-core/src/components/LabelWithOptions/LabelWithOptions.st.css]: 👍

File: packages/wix-ui-core/src/components/LabelWithOptions/LabelWithOptions.tsx (hunk 5 above)
Comment (shlomitc) [reply packages/wix-ui-core/src/components/LabelWithOptions/LabelWithOptions.tsx]: still needed here
